test/discord.test.ts

```diff
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
+                .withTasksInListForProject(jest.fn(() => Promise.resolve(returnedTasks)))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn().mockReturnValue(projectId))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withClient(client)
+                .withMappingController(mappingControllerMock)
+                .withCommandController(commandControllerMock)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent(`!tasks in ${taskList}`)
+                .withSend(jest.fn(async value => {
+                    if (value !== returnedTasks) {
+                        return;
+                    }
+
+                    expect(commandControllerMock.tasksInListForProject)
+                        .toBeCalledWith(taskList, projectId);
+                    done();
+                    return;
+                }))
+                .build();
+
+            client.emit('message', message);
+        });
+
+        it('should send warning when channel type is not text', () => {
+            const client = setupClient();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .withTasksInListForProject(jest.fn(() => Promise.resolve(new RichEmbed())))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withCommandController(commandControllerMock)
+                .withClient(client)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent(`!tasks in list`)
+                .withChannelType('voice')
+                .build();
+
+            client.emit('message', message);
+
+            expect(message.channel.send)
+                .toBeCalledWith('!tasks in list command must be called from' 
+                    + ' a text channel associated with a project');
+        });
+
+        it('should send warning and log error when error getting project ID', done => {
+            const client = setupClient();
+
+            const projectId = 0;
+            const error = 'error';
+            const list = 'List';
+            const channelName = 'channel';
+
+            let sentMessageValue = '';
+
+            const loggerMock = new LoggerMockBuilder()
+                .withWarn(jest.fn().mockImplementation(value => {
+                    expect(value).toBe(`Error getting tasks in ${list}: Error: ${error}`);
+                    expect(sentMessageValue).toBe(`Unable to find ActiveCollab project for channel ` 
+                        + channelName);
+                    done();
+                }))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn(() => { throw Error(error); }))
```

Comment:
See comment above about "Error" in Javascript. Since it's a constructor function you must call it with `new` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error 

Also can remove unnecessary parentheses